plugins/modules/identity/ipa/ipa_config.py

```diff
@@ -22,25 +22,141 @@
     description: Default e-mail domain for new users.
     aliases: ["emaildomain"]
     type: str
+  ipadefaultprimarygroup:
+    description: Default group for new users.
+    aliases: ["primarygroup"]
+    type: str
+    version_added: '2.4.0'
+  ipagroupsearchfields:
+    description: A comma-separated list of fields to search in when searching for groups.
+    aliases: ["groupsearchfields"]
+    type: str
+    version_added: '2.4.0'
+  ipahomesrootdir:
+    description: Default location of home directories.
+    aliases: ["homesrootdir"]
+    type: str
+    version_added: '2.4.0'
+  ipamaxusernamelength:
+    description: Maximum length of usernames.
+    aliases: ["maxusernamelength"]
+    type: str
+    version_added: '2.4.0'
+  ipapwdexpadvnotify:
+    description: Notice of impending password expiration, in days.
+    aliases: ["pwdexpadvnotify"]
+    type: str
```

Comment:
Shouldn't this be `int`?

Reply:
Addressed in 060cd1f.  I went back and forth on this initially.  I think they should be int, but some of the other ipa modules take what 'seem' to be int's as str's instead.  I am fine with going with int natively.  (Example: ipa_pwpolicy).

Reply:
Maybe that's for historical reasons... No idea :)